### ISSUE TITLE
scvi: consider allowing categorical_covariates to be used in a layer without batch

### ISSUE THREAD

**Issue by sjfleming** [cellarium-ai/cellarium-ml#294]
**Area for improvement:**
Right now in the config file specifying an scvi encoder architecture, if you specify a `cellarium.ml.models.scvi.LinearWithBatch` layer, then it requires a `batch_nb` and will also make use of a `categorical_covariate_np` if given.  The only way to _not_ use "batch" information is to not use a `cellarium.ml.models.scvi.LinearWithBatch` layer, but to instead use a `torch.nn.Linear` layer.  However, the `torch.nn.Linear` layer does not make use of categorical covariates.

**Use case:**
Let's imagine we want to train a model whose encoder does not use "batch" (we do want to try this for CAS).  In this case, a user will come in with a new cell where the batch label is unknown (and also probably previously unseen).  So providing a "batch" label consistent with the training data is not possible.  However... depending on the scvi model, it might be possible to provide categorical covariates.  In particular I am thinking about an scvi model with `[assay, suspension_type]` as categorical covariates.  `assay` and `suspension_type` do seem to explain most of the dataset-to-dataset variance overall, which makes sense.  But these are observed/known values, and a new user _would_ have this information about their query cells.  The scvi model could probably make great use of this information for embedding the new cells in the latent space, if the information could be provided.  But right now, there is no way to specify an scvi architecture that will make use of `categorical_covariate_np` in the encoder, but not require a `batch_nb`.

**Thoughts on a fix:**
Maybe add `if bias is None` here? 
https://github.com/cellarium-ai/cellarium-ml/blob/d9f9341d71312e1aeff17435396ec4ec31029333/cellarium/ml/models/scvi.py#L97-L100

This would enable us to use a `cellarium.ml.models.scvi.LinearWithBatch` layer in the scvi architecture.  If we do this, we should probably rename `cellarium.ml.models.scvi.LinearWithBatch` to `cellarium.ml.models.scvi.LinearWithCovariates` or something.  Maybe we should actually have three classes to make it totally explicit: `LinearWithBatch`, `LinearWithBatchAndCovariates`, `LinearWithCovariates`?
